### PR TITLE
chore: upgrade to eth-ape-0.1.0a26

### LINKED ISSUE #8: chore: pin ape 0.1.0a23
pin should be updateable now that test suite no longer supports python 3.6

### PR DESCRIPTION
**Pull request by sabotagebeats**
fixes #8

### CODE REVIEW

**Comment by sabotagebeats** [comment]
Should this pin be higher?

File: setup.py [hunk 1]
```diff
@@ -61,7 +61,7 @@
     install_requires=[
         "importlib-metadata ; python_version<'3.8'",
         "py-solc-x>=1.1.0,<1.2.0",
-        "eth-ape>=0.1.0a15",
+        "eth-ape>=0.1.0a23",
```
Comment (unparalleled-js) [inline setup.py:64]: Ha just now seeing this, might as well switch to `0a25`

**Review by unparalleled-js — APPROVED** [review]
Can switch to `0a25` now

**Comment by sabotagebeats** [comment]
@unparalleled-js switched to 0a25

**Comment by sabotagebeats** [comment]
hmm the type check is failing?

**Comment by unparalleled-js** [comment]
Ohhh, I remember this issue.
This has to do with recent mypy upgrades.

Try adding this somewhere to the `pyproject.toml`:

```cfg
[tool.mypy]
exclude = "build/"
```

(See `ape` as an example if you need, I had to do this there for this same reason)

**Comment by sabotagebeats** [comment]
yikes now it's failing functional 😿

**Comment by sabotagebeats** [comment]
```
Run pytest -m "not fuzzing"
============================= test session starts ==============================
platform linux -- Python 3.9.7, pytest-6.2.5, py-1.11.0, pluggy-0.13.1
rootdir: /home/runner/work/ape-solidity/ape-solidity, configfile: pyproject.toml, testpaths: tests
plugins: hypothesis-6.24.2, forked-1.3.0, xdist-2.4.0, eth-ape-0.1.0a26, web3-5.24.0, cov-3.0.0
gw0 I / gw1 I
INTERNALERROR> def worker_internal_error(self, node, formatted_error):
INTERNALERROR>         """
INTERNALERROR>         pytest_internalerror() was called on the worker.
INTERNALERROR>     
INTERNALERROR>         pytest_internalerror() arguments are an excinfo and an excrepr, which can't
INTERNALERROR>         be serialized, so we go with a poor man's solution of raising an exception
INTERNALERROR>         here ourselves using the formatted message.
INTERNALERROR>         """
INTERNALERROR>         self._active_nodes.remove(node)
INTERNALERROR>         try:
INTERNALERROR> >           assert False, formatted_error
INTERNALERROR> E           AssertionError: Traceback (most recent call last):
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/_pytest/main.py", line 269, in wrap_session
INTERNALERROR> E                 session.exitstatus = doit(config, session) or 0
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/_pytest/main.py", line 322, in _main
INTERNALERROR> E                 config.hook.pytest_collection(session=session)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR> E                 return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR> E                 return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR> E                 self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR> E                 return outcome.get_result()
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR> E                 raise ex[1].with_traceback(ex[2])
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR> E                 res = hook_impl.function(*args)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/_pytest/main.py", line 333, in pytest_collection
INTERNALERROR> E                 session.perform_collect()
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/_pytest/main.py", line 641, in perform_collect
INTERNALERROR> E                 hook.pytest_collection_finish(session=self)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR> E                 return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR> E                 return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR> E                 self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/callers.py", line 203, in _multicall
INTERNALERROR> E                 gen.send(outcome)
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/ape_test/runners.py", line 92, in pytest_collection_finish
INTERNALERROR> E                 self.networks.active_provider.connect()
INTERNALERROR> E               File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/ape_geth/providers.py", line 162, in connect
INTERNALERROR> E                 raise GethNotInstalledError()
INTERNALERROR> E             ape_geth.providers.GethNotInstalledError: geth is not installed and there is no local provider running.
INTERNALERROR> E             Things you can do:
INTERNALERROR> E             	1. Install geth and try again
INTERNALERROR> E             	2. Run geth separately and try again
INTERNALERROR> E             	3. Use a different ape provider plugin
INTERNALERROR> E           assert False
INTERNALERROR> 
INTERNALERROR> /opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/xdist/dsession.py:187: AssertionError
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/_pytest/main.py", line 269, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/_pytest/main.py", line 323, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/callers.py", line 203, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/xdist/dsession.py", line 112, in pytest_runtestloop
INTERNALERROR>     self.loop_once()
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/xdist/dsession.py", line 135, in loop_once
INTERNALERROR>     call(**kwargs)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/xdist/dsession.py", line 175, in worker_workerfinished
INTERNALERROR>     self._active_nodes.remove(node)
INTERNALERROR> KeyError: <WorkerController gw0>

============================ no tests ran in 1.50s =============================
Error: Process completed with exit code 3.
```

File: setup.py (hunk 1 above)
Comment (sabotagebeats) [reply setup.py]: I did switch to 25 but now i'm having issues passing pytest check

**Comment by sabotagebeats** [comment]
gonna try `26` for fun :P

**Comment by sabotagebeats** [comment]
looks like same error with `0.1.0a26`

**Comment by unparalleled-js** [comment]
If you use 26, you may need to add this`-p no:ape_test` to the `addopts` part of `pyproject.toml` under the stanza `[tool.pytest.ini_options]`.

**Comment by unparalleled-js** [comment]
@sabotagebeats I just tested it out in another branch, adding will cause a passing build`-p no:ape_test`
See #12 as an example!

**Comment by sabotagebeats** [comment]
hooray, passing!

**Pull request merged**